modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,57 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
+				var image         = media[ i ],
+					media_details = image.media_details,
+					media_sources = media_details.sources,
+					sizes         = media_details.sizes,
+					sizes_keys    = Object.keys( sizes ),
+					images        = document.querySelectorAll( 'img.wp-image-' + image.id );
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					var src    = images[ j ].src,
+						srcset = images[ j ].getAttribute( 'srcset' );
+
+					// If there are no sizes, then replace src through sources, there is nothing more to replace.
+					if ( media_sources ) {
+						src = src.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
+						images[ j ].setAttribute( 'src', src );
+						break;
+					}
+
+					// If the full image has no JPEG version available, no sub-size will have JPEG available either.
+					if ( ! sizes.full.sources['image/jpeg'] ) {
+						continue;
+					}
```

Comment:
This check is not in the right place here: It isn't based on `images` at all, so we should check that right away, above the `images` loop: If the full size has no `image/jpeg` version available, then we don't even need to attempt replacement for this `image`.

Even better would be: Let's check this even before the `document.querySelectorAll` call, because if there is no JPEG, we don't even need to query the image elements for this attachment.

Reply:
The https://github.com/WordPress/performance/pull/443/commits/c2a93ba3383381d3658ad28219b00a066639e892 address this feedback.